karma.config.js

```diff
@@ -98,7 +107,14 @@ module.exports = function (config) {
 
     // start these browsers
     // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
-    browsers: ['Chrome'],
+    browsers: ['puppeteer'],
+
+    puppeteerLauncher: {
+      puppeteer: {
+        headless: config.headless || false,
```

Comment:
[nit]: could this be just:

headless: Boolean(config.headless) or
headless: !!config.headless